// grunt-stamplay/Gruntfile.js
/*
 * grunt-stamplay
 * https://github.com/Stamplay/grunt-stamplay
 *
 * Copyright (c) 2016 Mirko Di Serafino
 * Licensed under the MIT license.
 */

'use strict';

module.exports = function defineTask(grunt) {
  grunt.initConfig({
    stamplay: {
      options: {
        appId: 'YOUR-APPID',          // appId of the project
        apiKey: 'YOUR-APIKEY',        // apiKey of the project
        public: './'                  // public folder of your project, must contain index.html
      },
      deploy: {
        message: 'deploy message',    // message to describe the deploy
        ignore: [                     // files that mustn't be uploaded with the deploy
          '**/.*',
          '**/node_modules/**'
        ],
        headers: [{                   // describes cache headers for the current deploy
          source: '**/*.@(html)',
          headers: [{
            key: 'Cache-Control',
            value: 'max-age=7200'
          }]
        }]
      }
    }
  });

  grunt.loadTasks('tasks');
  grunt.registerTask('default', ['stamplay']);
};
